### PR TITLE
Ignite-5798

### PR DESCRIPTION
**Pull request by 1vanan**
Logging Ignite configuration at startup

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 1]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
     }
 
     /**
+     *Log fields and values of IgniteConfiguration class.
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: * Logs the fields and the values of the used IgniteConfiguration instance.

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 2]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
...
+     */
+    private void ackIgniteConfigurationInfo(){
+
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: No new line required.

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 3]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
...
+        Field[] allFields = IgniteConfiguration.class.getDeclaredFields();
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: allFields -> fields

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 4]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
...
+
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: No new line required.

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 5]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
...
+
+        if (log.isInfoEnabled()) {
+            for (Field field :
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: One line style.

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 6]
```diff
@@ -810,6 +811,7 @@ public void start(
         ackCacheConfiguration();
         ackP2pConfiguration();
         ackRebalanceConfiguration();
+        ackIgniteConfigurationInfo();
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: Let's call it after "ackConfigUrl();"

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 7]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
...
+    private void ackIgniteConfigurationInfo(){
+
+        Field[] allFields = IgniteConfiguration.class.getDeclaredFields();
+
+
+        if (log.isInfoEnabled()) {
+            for (Field field :
+                    allFields) {
+                try {
+                    log.info(field.getName() + ": " + field.get(cfg));
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: Let's define output format.
String.format("%s : %s", field.getName(), field.get(cfg));
Your variants are welcome.

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 8]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
...
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: throw new IgniteException("Failed to access field for getting value.", e);

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 9]
```diff
@@ -2503,6 +2505,37 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
...
+                }
+            }
+        }
+
+        if (log.isQuiet()) {
+            for (Field field :
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: Look the comments above.

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java (hunk 5 above)
Comment (1vanan) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: What does it mean? After writing code, i always do autoformatting in Idea and this is standard format.

Comment (daradurvs) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: It means:
for (Field field : fields) {}
Please, use projects code style, import to IDE: ignite\idea\ignite_codeStyle.xml

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 10]
```diff
@@ -2503,6 +2505,35 @@ private void ackRebalanceConfiguration() throws IgniteCheckedException {
     }
 
     /**
+     *Logs the fields and the values of the used IgniteConfiguration instance.
```
Comment (daradurvs) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: Missed space after "*".

**Pull request closed without merging**